types/tpdirect/tpdirect-tests.ts

```diff
@@ -28,9 +28,9 @@ TPDirect.card.getTappayFieldsStatus();
 TPDirect.card.onUpdate((update) => {
     update.cardType; // $ExpectType "mastercard" | "visa" | "jcb" | "amex" | "unionpay" | "unknown"
     update.hasError; // $ExpectType boolean
-    update.status.ccv; // $ExpectType 0 | 1 | 2 | 3
-    update.status.number; // $ExpectType 0 | 1 | 2 | 3
-    update.status.expiry; // $ExpectType 0 | 1 | 2 | 3
+    update.status.ccv; // $ExpectType 0 | 1 | 2 | 3 || 0 | 2 | 1 | 3
+    update.status.number; // $ExpectType 0 | 1 | 2 | 3 || 0 | 2 | 1 | 3
+    update.status.expiry; // $ExpectType 0 | 1 | 2 | 3 || 0 | 2 | 1 | 3
```

Comment:
I tried, and it was far too expensive during .d.ts emit =(

Reply:
:( I mean, maybe we can do something like give these types special type ids that sort them during union construction, rather than assigning them on a first-come-first-served basis? AFAIK we could change type ids to strings without affecting much.

Reply:
could you maybe just sort them for display purposes? this shouldn't affect compiler perf but would make such "snapshots" as here more stable

Reply:
It does hurt performance, because the project build system needs to generate .d.ts files during builds.

Reply:
@weswigham It would probably work to allocate some space for numbers, at least. I can't remember whether integer representation is signed, but if so, we could just reserve -20_000 to -10_000 for numbers 0 to 10,000.

Reply:
We could associate a `sortId` with all types that we sort on before base `Id` and allow `sortId` to be a string, so we can sort somewhat arbitrarily. Only issue is that right now we have some hardcoded assumptions that `undefined`, if present in a union, will always be the first element (which we then always swap to last upon display!) as an optimization. Designing sort IDs to maintain that will be fun. Maybe `undefined` gets the vaunted empty string sort ID?